model_one/cli.py

```diff
@@ -1,198 +1,274 @@
+
+        if train_iters:
+            model["model_params"] = {"train_iters": train_iters}
+        
+        return cls.create(model)
+
+    @classmethod
+    def models(cls) -> List['BeyondmlModel']:
+        r = BeyondmlAPI.models()
+
+        return [
+            cls.from_dict(data) for data in r.get("models", [])
+        ]
+
+    @classmethod
+    def load(cls, filename: str) -> 'BeyondmlModel':
+        with open(filename, "r") as fl:
+            data = json.load(fl)
+            return cls.from_dict(data)
+
+    @inited
     def save(self, filename):
-        with open(filename, 'w') as fl:
+        with open(filename, "w") as fl:
             json.dump({
-                'model_name': self._id,
-                'status': self._status,
-                'model_type': self._model_type,
+                "model_name": self._id,
+                "status": self._status,
+                "model_type": self._model_type,
             }, fl)
 
-    def _upload_fit(self, train_X, train_y, validate_X, validate_y):
-        self.status()
-        if self._status == 'Created':
+    @property
+    def is_initied(self) -> bool:
+        return self._id is not None
+    
+    @property
+    @inited
+    def is_ready(self):
+        return self.status is BeyondmlModelStatus.READY
+    
+    @property
+    @inited
+    def status(self) -> BeyondmlModelStatus:
+        r = BeyondmlAPI.status(self._id)
+        self._status = status = r["status"]
+        return BeyondmlModelStatus(status.lower())
+
+    @property
+    @inited
+    def type(self) -> BeyondmlModelType:
+        return BeyondmlModelType(self._model_type.lower())
+
+    @inited
+    def fit(
+        self,
+        train_X: Union[list, Series, None], 
+        train_y: Union[list, Series, None],
+        validate_X: Union[list, Series, None],
+        validate_y: Union[list, Series, None]
+    ):
+        if all(train_X, train_y, validate_X, validate_y):
             self.upload(train_X, train_y, validate_X, validate_y)
-
-        if self._status == 'DatasetsLoaded':
-            self.fit()
-        return self
-
-
-def load(filename):
-    with open(filename, 'r') as fl:
-        return BeyondmlModel(json.load(fl))
-
-
-def _create(data):
-    headers = _get_auth_header()
-    headers['Content-Type'] = 'application/json'
-    r = requests.post(_models_api(), headers=headers, data=json.dumps(data))
-    if r.status_code != 200:
-        raise Exception(r.text)
-    return BeyondmlModel(r.json())
-
-
-def _create_or_load(data: dict, filename: str = None):
-    if filename is not None and os.path.isfile(filename):
-        model = load(filename)
-        if model._model_type != data['model_type']:
-            raise Exception(
-                'Model in the file is not {}'.format(data['model_type']))
-        return model
-    m = _create(data)
-    m.save(filename)
-    return m
-
-def create_generator(filename: str = None, train_iters : int = None):
```

Comment:
These functions do a not-so-good thing by allowing creation of a non-generator and non-classifier models by loading them from provided files